Atmos fixes and tweaks.

MrsTonedOne opened the pull request:
So basically, one pr caused a bug where firelocks would not let atmos pass thru them when open, to fix it, I had to re-fix the bug that pr was fixing, fixing that caused a bug that hadn't been known because the aforementioned fixed bug broke pipenet code in a minor and not immediately noticeable way, this chain continued until FINALLY I was able to fix firedoors, 7 hours later.

Anywho, Atom as fuck changes:

Mostly reverted #9896 for causing #10828
Moved pipenet setup out of obj/initialize and into its own proc called from MC's initialize.
Removed obj/initialize anything in atmos machinery.
Fixed a bug in atmospherics/components/Destroy() that prevented garbage collection of nearby nodes (again)
Fixed a bug in atmospherics/components/disconnect() that would prevent garbage collection of recently disconnected pipes as well as make pipenet code think they were still connected.
Fixed a bug in atmospherics/components/disconnect() that would null out a pipenet that was still valid and in use.
Fixed a bug in atmospherics/components/nullifyPipenet() that would prevent a pipenet from getting nulled out and garbage collected.
Fixed a bug in atmospherics/components/pipeline_expansion() that would cause it to return the letter "n" rather than a node.
Fixed a bug in atmospherics/components/pipeline_expansion() that would cause it to return all of its nodes rather than the node connected to the given pipenet.
Fixed a bug in atmospherics/components/pipeline_expansion() that would cause it to return null rather than a null list if the node wasn't connected.
Fixed a bug in atmospherics/components/replacePipenet() where it didn't actually do anything because it was assigning the new pipenet to the typecasted local var, not the object's pipenet list.
Reverted my change in #10593 to atmospherics/components/proc/update_parents() because I fixed all the bugs that made it necessary (hopefully)
Fixed a bug in pipe meters where they would keep a reference to their pipe (preventing the pipe from qdeling) if they were pulled by a singularity before their attached pipe was.

Fixes #10682 
Fixes #10753
Fixes #10828

MrsTonedOne committed:
Removes MC's after init proc

It was only used for air, we can just move air to init last with priority = -1 (this is what this is for)

MrsTonedOne committed:
Moves build_network to air's initialize, Removes initialize() from atmos machinery

MrsTonedOne committed:
Fixes some pipenet bugs in components_base

MrsTonedOne committed:
Fixes objs that block air breaking atmos when spawned in a non-blocking state
This included any door that started open (firedoors being the main example)
Fixes #10828

Aranclanos commented:
bretty gud

duncathan commented:
:+1: fixed my mistakes, and then some

MrPerson commented on code/ATMOSPHERICS/components/components_base.dm:
```diff
@@ -199,7 +195,6 @@ Helpers
\ No newline at end of file
+	for(var/I = 1; I <= device_type; I++)
+		var/datum/pipeline/parent = parents["p[I]"]
+		parent.update = 1
```
Not sure why you changed this considering for(var/P in parents) is twice as fast as for(var/I = 1; I <= device_type; I==)

Course it probably doesn't matter either, it's just a weird change

duncathan replied:
that was just him putting it back to how I'd done it
the reason I have all the loops like that is basically just for consistency
for a lot of the loops it is necessary (the ones before the lists are initialized, the ones where there's multiple lists being edited) and for the ones for which it isn't necessary, I thought it'd be nice to just let it all be consistent

MrStonedOne replied:
>Reverted my change in #10593 to atmospherics/components/proc/update_parents() because I fixed all the bugs that made it necessary (hopefully)

Aran requested I do that.

Aranclanos replied:
its pretty good at runtiming like hell when there's a bug, so it can stay temporally(TM)
plus consistency with the rest of these procs under this section

Cheridan commented:
It's so weird to see atmos development

MrsTonedOne committed:
Minor fixes to air

It was initializing turfs after atmos machinery but some of those machines expect turfs to be initialized before they initialize.

Also fixes it clearing all active turfs and not just the one in the specified zlevel in turf initialization.

MrsTonedOne committed:
fixes edge case with meters and the singularity

If a pipe meter was singularity_pull'ed before its attached pipe, the attached pipe would not disconnect itself from the meter since the pipe checks only its own turf on deconstruction for meters.
